rust/src/countme/repo.rs

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+use anyhow::{Context, Result};
+use std::fs;
+use std::path::PathBuf;
+
+use ini::Ini;
+
+/// Location for DNF repositories configuration
+pub const YUM_REPOS_D: &str = "/etc/yum.repos.d";
+
+/// Repository configuration
+#[derive(Debug)]
+pub struct Repo {
+    name: String,
+    enabled: bool,
+    count_me: bool,
+    meta_link: String,
+}
+
+/// Read all repository configuration files from the default location
+pub fn all() -> Result<Vec<Repo>> {
+    let configs = fs::read_dir(YUM_REPOS_D)
+        .with_context(|| format!("Could not list files in: {}", YUM_REPOS_D))?;
+    let mut repos = Vec::new();
+    for c in configs {
+        let path = match c {
+            Err(e) => {
+                println!("Could not list file in : {}: {}", YUM_REPOS_D, e);
+                continue;
+            }
+            Ok(p) => p.path(),
+        };
+        match from_file(&path) {
+            Err(e) => {
+                println!("Skipped file: {}: {}", path.display(), e)
+            }
+            Ok(mut r) => repos.append(&mut r),
+        }
+    }
+    Ok(repos)
+}
+
+/// Read repository configuration from a file
+fn from_file(path: &PathBuf) -> Result<Vec<Repo>> {
+    let i = Ini::load_from_file(path).with_context(|| "Could not parse as INI".to_string())?;
+    let mut repos = Vec::new();
+    for (sec, prop) in i.iter() {
+        let mut repo = match sec {
+            None => {
+                println!("Skipping config values not attached to a repository");
+                continue;
+            }
+            Some(s) => Repo {
+                name: String::from(s),
+                enabled: false,
+                count_me: false,
+                meta_link: "".to_string(),
+            },
+        };
+        for (k, v) in prop.iter() {
+            match k {
+                "countme" => {
+                    if let "1" = v {
```

Comment:
I think technically `"true"` is valid too here. See https://gitlab.gnome.org/GNOME/glib/-/blob/f22b136b114671375778574161f9ded136518709/glib/gkeyfile.c#L4549-4552. Heh, looks like libdnf is even more permissive: https://github.com/rpm-software-management/libdnf/blob/45981d5f53980dac362900df65bcb2652aa8d7c7/libdnf/conf/OptionBool.hpp#L30-L31.

Reply:
Maybe we should use serde+YAML to deserialize it as a `bool`?  (Not kidding actually)

Reply:
Will that work with 0/1 since in YAML those are integers? Meh, probably just matching glib and accepting 0/1/true/false is good enough.